app/models/grda_warehouse/confidence/base.rb

```diff
@@ -98,15 +98,15 @@ def self.to_csv
     # gives the option to create for one client
     def self.setup_for_client client_id
       collections = collection_dates_for_client(client_id)
-      new.insert_batch(self, collections.first.keys, collections.map(&:values))
+      import(collections.first&.keys, collections.map(&:values))
```

Comment:
This should probably have a `return unless collections.present?` before line 101, instead of the safe navigator.

Reply:
✅